pom.xml

```diff
@@ -1649,10 +1666,25 @@
       </activation>
       <properties>
         <hbase.profile>2.5</hbase.profile>
-        <hbase.compat.version>2.5.0</hbase.compat.version>
+        <hbase.compat.version>2.5.4</hbase.compat.version>
         <hadoop.version>3.2.3</hadoop.version>
         <hbase.version>${hbase-2.5.runtime.version}</hbase.version>
-        <tephra.hbase.compat.version>2.4</tephra.hbase.compat.version>
+      </properties>
+    </profile>
+    <profile>
+      <!-- This WILL work with the public -hadoop3 artifacts -->
+      <id>phoenix-hbase-compat-2.5.0</id>
+      <activation>
+        <property>
+          <name>hbase.profile</name>
+          <value>2.5.0</value>
+        </property>
+      </activation>
+      <properties>
+        <hbase.profile>2.5.0</hbase.profile>
+        <hbase.compat.version>2.5.0</hbase.compat.version>
```

Comment:
shall we keep this also `2.5.4` just to maintain same compat version as before?

Reply:
The convention we have used so far is that the profile without patch level, i.e. 2.5 always points to the latest supported HBase patch level (and corresponding compatibility module) within that minor version.
When we need to add a new profile, we move the patch level-less profile to point to the latest one, and create a new profile which is named after the first patch level it supports.

Before this patch we had the 2.5 profile which used the existing 2.5.0 compatibility moudule.
Now the 2.5 profile depends on the new 2.5.4 compatibility module instead, and we've added the new 2.5.0 profile, which depends on the old 2.5.0 compatibility module.

Yes, this means that you have to change the profile if you keep using an older HBase version, but 2.5 profile always works with the latest 2.5 release. We can't have it both ways unfortunately.

We could add an explicit 2.5.4 profile as an alias to 2.5, but I don't think that would improve things.